### PR TITLE
support embedding scene data to PNG/SVG

### PR DESCRIPTION
**Pull request by dwelle**
fixes https://github.com/excalidraw/excalidraw/issues/414

- [x] PNG
- [x] SVG 

(note: the libs are tiny but I'm dynamically importing them because webpack unnecessarily bundles in Node's Buffer library :frowning_face:)

image

### CODE REVIEW

**Comment by dwelle** [comment]
@tomayac is it possible to save a png with a custom mimeType (or something) which we'd use as a filter when opening files so that you can't open just any png file (just for UX)?

**Comment by tomayac** [comment]
> @tomayac is it possible to save a png with a custom mimeType (or something) which we'd use as a filter when opening files so that you can't open just any png file (just for UX)? (I have a feeling I asked you once already in similar context 😄)

(Sorry for the late reply, I was on vacation.) I doubt that this (i.e., the filtering part) would work in practice. I guess the easiest would be to use a separate extension.

**Comment by dwelle** [comment]
(No problem.) Yea, that's not gonna fly with `.png`/`.svg` :P.

File: src/base64.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// `btoa(unescape(encodeURIComponent(str)))` hack doesn't work in edge cases and
+//  `unescape` API shouldn't be used anyway.
```
Comment (lipis) [inline src/base64.ts:2]:
```suggestion
// `unescape` API shouldn't be used anyway.
```

**Review by lipis — APPROVED** [review]
It just feels that this needs some more explanation.. maybe a link to how to or a blog post describing what it does.. it's quite useful in some cases!

**Comment by dwelle** [comment]
There's an extended tooltip, but people in general don't know about hover tooltips. I was thinking of adding a :information_source: button next to the checkbox that would show our own tooltip implem (the one we're using for the shield button). But maybe for another PR.

**Comment by lipis** [comment]
Yes an icon or just some text with a link that indicates something.. maybe our own tooltip should be enough.

**Comment by frantic** [comment]
Amazing, thanks for making it happen 🎉🎉

Small ask: since this can make the file a little bigger, would be nice to have something like (+5KB) next to this option :)

**Comment by dwelle** [comment]
> Small ask: since this can make the file a little bigger, would be nice to have something like (+5KB) next to this option :)

:+1: I was thinking of the same thing, I'll create an issue for it

**Review by Rada8732 — APPROVED** [review]

**Comment by Rada8732** [comment]
Try both

**Pull request merged**